include/xtensor/xbuilder.hpp

```diff
@@ -267,6 +267,84 @@ namespace xt
             }
         };
 
+        // These methods should be private methods of linspace_generator, however thi leads
+        // to ICE on VS2015
+        template <class R, class E, class U, class X, XTL_REQUIRES(xtl::is_integral<X>)>
+        inline void linspace_assign_to(xexpression<E>& e, U start, U, X step, bool) noexcept
+        {
+            arange_assign_to(e, start, step);
+        }
+
+        template <class R, class E, class U, class X, XTL_REQUIRES(xtl::negation<xtl::is_integral<X>>)>
+        inline void linspace_assign_to(xexpression<E>& e, U start, U stop, X step, bool endpoint) noexcept
+        {
+            auto& buf = e.derived_cast().storage();
+            using size_type = decltype(buf.size());
+            using mult_type = get_mult_type<U, X>;
+            size_type num = buf.size();
+            for(size_type i = 0; i < num; ++i)
+            {
+                buf[i] = static_cast<R>(start + step * mult_type(i));
+            }
+            if (num > 0 && endpoint)
+            {
+                buf[num-1] = static_cast<R>(stop);
```

Comment:
Here is the fix.

Reply:
I guess using the same mechanism as in `arange_to` (adding step to the previous computed value instead of performing the whole computation) leads to the same potential rounding errors? (because `start + step + step` might be slightly different from `start + step * 2` for instance)

Reply:
@JohanMabille Thanks for the review! But I don't understand the comment here.  `stop` here is the original input, which is not a calculated value.